// react-static-plugin-mdx/node.api.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && Symbol.iterator in Object(iter)) return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

var path = require('path');

var _default = function _default(_ref) {
  var _ref$includePaths = _ref.includePaths,
      includePaths = _ref$includePaths === void 0 ? [] : _ref$includePaths,
      _ref$extensions = _ref.extensions,
      extensions = _ref$extensions === void 0 ? ['.md', '.mdx'] : _ref$extensions,
      _ref$mdxOptions = _ref.mdxOptions,
      mdxOptions = _ref$mdxOptions === void 0 ? {} : _ref$mdxOptions,
      _ref$parseFrontMatter = _ref.parseFrontMatter,
      parseFrontMatter = _ref$parseFrontMatter === void 0 ? false : _ref$parseFrontMatter;
  return {
    afterGetConfig: function afterGetConfig(_ref2) {
      var config = _ref2.config;
      config.extensions = [].concat(_toConsumableArray(config.extensions), _toConsumableArray(extensions));
    },
    webpack: function webpack(webpackConfig, _ref3) {
      var defaultLoaders = _ref3.defaultLoaders;

      var mdxLoaderPath = require.resolve('@mdx-js/loader');

      webpackConfig.module.rules[0].oneOf.unshift({
        test: /.mdx?$/,
        include: [defaultLoaders.jsLoader.include].concat(_toConsumableArray(includePaths)),
        use: [defaultLoaders.jsLoader.use[0], {
          loader: mdxLoaderPath,
          options: mdxOptions
        }, parseFrontMatter && path.join(__dirname, './loader/fm-loader')].filter(function (x) {
          return x;
        }) // Remove falsy value when not parsing front matter

      });
      return webpackConfig;
    }
  };
};

exports["default"] = _default;
